Assets/Scripts/UI/PepemonFactoryCardCache.cs

```diff
@@ -0,0 +1,90 @@
+using Nethereum.Unity;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using UnityEngine;
+using UnityEngine.Networking;
+
+/// <summary>
+/// Downloads and caches PepemonFactory card information, including metadata 
+/// and textures
+/// </summary>
+class PepemonFactoryCardCache
+{
+    // TODO: Implement local cache using Application.persistentDataPath
+
+    private const ulong MAX_TOKEN_ID = ulong.MaxValue;
+    private static Dictionary<ulong, Texture2D> cardTextures = new Dictionary<ulong, Texture2D>();
+    private static Dictionary<ulong, PepemonFactory.CardMetadata> cardMetadata = new Dictionary<ulong, PepemonFactory.CardMetadata>();
+
+    public async Task PreloadAll()
+    {
+        await PreloadAllMetadata();
+        await PreloadAllImages();
+    }
+
+    private async Task PreloadAllMetadata()
+    {
+        // Iterate through card IDs and store the card metadata
+        // No way to get the maximum token id, so iteration is
+        // complete when an error is thrown.
+        for (ulong i = 0; i < MAX_TOKEN_ID; i++)
+        {
+            try
+            {
+                cardMetadata[i] = await PepemonFactory.GetCardMetadata(i);
+            }
+            catch (Exception e)
+            {
+                Debug.Log($"Preloading finished at tokenId: {i} with error: {e.Message}");
```

Comment:
```suggestion
                Debug.Log($"Preloading finished at tokenId: {i} with error: {e.Message}");
                break;
```
there should be an exit for this loop, otherwise it will continue trying and failing